src/box/wal.c

```diff
@@ -1319,9 +1319,8 @@ wal_write(struct journal *journal, struct journal_entry *entry)
 	if (wal_write_async(journal, entry) != 0)
 		return -1;
 
-	bool cancellable = fiber_set_cancellable(false);
-	fiber_yield();
-	fiber_set_cancellable(cancellable);
+	while (!entry->complete)
+		fiber_yield();
```

Comment:
First check will always fail because the entry doesn't have a chance to complete until a yield. I would change it to:
```C
assert(!entry->complete);
do {
    fiber_yield();
} while (!entry->complete);
```
or even:
```C
assert(!entry->complete);
do {
    fiber_yield();
} while (unlikely(!entry->complete));
```
if you are a fan.

Reply:
Done